src/carve/impl.clj

```diff
@@ -271,3 +273,74 @@
                          (dissoc analysis :unused-var-refers)))
                 (reportize results)))
           (reportize results))))))
+
+(set! *warn-on-reflection* true)
+(s/check-asserts true)
```

Comment:
Not sure if `check-asserts` will be limited to the current ns?

Reply:
This is a global flag, but good point. I think we should only set this in the main ns (which is intended to be used from the CLI / native). Setting this as part of a library isn't a good idea, this choice should be done by the user.